### PR TITLE
Move ``Run`` and ``PyLinter`` options to separate file

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

### CODE REVIEW

File: pylint/lint/pylinter.py [hunk 1]
```diff
@@ -622,8 +261,7 @@ def __init__(
         self.stats = LinterStats()
 
         # Attributes related to (command-line) options and their parsing
-        self._external_opts = options
```
Comment (Pierre-Sassoulas) [inline pylint/lint/pylinter.py:625]: This was not used anywhere?

**Review by Pierre-Sassoulas — COMMENTED** [review]
I think it's a pretty nice step in the right direction of moving things away from the PyLinter. I'm only wondering if ``base_option.py`` should be in ``pylint.config`` instead.

File: pylint/lint/pylinter.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/lint/pylinter.py:625]: No! I double-checked. It's private so we can remove it without any hassle, the first time I think 😄

**Comment by DanielNoord** [comment]
> I think it's a pretty nice step in the right direction of moving things away from the PyLinter. I'm only wondering if `base_option.py` should be in `pylint.config` instead.

I'll let you decide. I was doubting this as well. I thought "options of checkers are also in the same file so it is probably better to keep the options together with the object that provides them".

**Review by Pierre-Sassoulas — APPROVED** [review]
> keep the options together with the object that provides them".

Make sense, let's keep it that way than !

**Pull request merged**